rails/client/app/bundles/kaiju/components/Component/utilities/dispatcher.js

```diff
@@ -302,6 +314,59 @@ const registerDispatcher = (store, root) => {
     }
   };
 
+  /**
+   * Checks if the coordinate is within the selected component.
+   * @param {float} x - The x coordinate.
+   * @param {float} y - The y coordinate.
+   * @return {boolean} - true if the coordinate is within the selected component.
+   */
+  const isInside = (x, y) => {
+    const selectedComponent = findComponentById(getSelectedComponent());
+
+    if (!selectedComponent) {
+      return false;
+    }
+
+    const { left, right, top, bottom } = selectedComponent.getBoundingClientRect();
+    if (x >= left && x <= right && y >= top && y <= bottom) {
+      return true;
+    }
+
+    return false;
+  };
```

Comment:
This could potentially be replaced with a recursive function that travels up the tree using the parent attribute of each component from the store.